lib/engine/hex.rb

```diff
@@ -143,7 +143,7 @@ def lay(tile)
               next unless ability.hex == coordinates
 
               ability.tile = new_city.tile
-              ability.city = new_city.index
+              ability.city = new_city.tile.cities.find_index(new_city)
```

Comment:
Why is this necessary?

Reply:
This is a bug fix. new_city.index is the index value from the list of tile parts, not the list of cities. These might match if all the cities are defined at the start of the part list, but I had a tile where the definition was "city, path, city, path, city, part…" This was giving a new_city index of 4, when there were only three cities on the tile. I could have just changed the order of the parts on this tile, but I thought it would be better to fix the bug so that it didn't catch anyone else out.

Reply:
I've created a separate pull request for this change, #8613.

Reply:
This pull request has been merged.